### PR TITLE
Improve browsing experience

### PR DESCRIPTION
**Pull request by dichotommy**
Since [removing all site READMEs](github.com/meilisearch/documentation/issues/1293) may [take some more time](github.com/meilisearch/documentation/pull/1400), this PR bundles several non-invasive improvements to browsing experience, such as:

- Add sidebar to the FAQ (replaces table of contents)
- Change Learn and Reference navbar links to point to the quick start and API reference readme, respectively
- Switch the order of API references and feature references in the `/reference` sidebar

### CODE REVIEW

File: resources/faq.md [hunk 1]
```diff
@@ -1,15 +1,13 @@
 ---
 permalink: /faq.html
+sidebar: auto
 ---
 
 # FAQ
 
 This FAQ is still a work in progress.
```
[the remark this answers is not in the document]
Comment (dichotommy) [reply resources/faq.md:8]: Good point, let's do it.

Comment (dichotommy) [inline resources/faq.md:8]:
```suggestion
This FAQ is created in collaboration with our users.
```

File: resources/faq.md [hunk 2]
```diff
@@ -1,15 +1,13 @@
...
-If you have any questions we want to hear from you. Your feedback will help us improve this page!
 
-#### Table of contents
-
-[[toc]]
+If you have any questions, we want to hear from you! Your feedback will help us improve this page.
```
Comment (dichotommy) [inline resources/faq.md:10]:
```suggestion
If you have any questions that aren't answered here, we want to hear from you! Your feedback will help us improve our documentation.
```

**Review by guimachiavelli — APPROVED** [review]
✨

**Comment by dichotommy** [comment]
bors merge

**Pull request merged**